lib/src/Base/Optim/OptimizationProblemImplementation.cxx

```diff
@@ -97,6 +93,12 @@ NumericalMathFunction OptimizationProblemImplementation::getObjective() const
 
 void OptimizationProblemImplementation::setObjective(const NumericalMathFunction & objective)
 {
+  // clear constraint, bounds
+  LOGWARN(OSS() << "Clearing constraints and bounds");
+  equalityConstraint_ = NumericalMathFunction();
+  inequalityConstraint_ = NumericalMathFunction();
+  bounds_ = Interval(0);
+
   clearLevelFunction();
```

Comment:
The constraints should be removed only if the new objective function has an incompatible dimension. It allows eg to switch to a finer discretized robust objective function while keeping the constraints as is.